pkg/api/vlabs/validate.go

```diff
@@ -407,6 +407,12 @@ func (a *Properties) validateAgentPoolProfiles(isUpdate bool) error {
 			}
 		}
 
+		if to.Bool(agentPoolProfile.EnableVMSSNodePublicIP) {
```

Comment:
let's add this case to unit tests as well

Reply:
Sure, done.